crates/p2p/src/swarm/mod.rs

```diff
@@ -571,44 +574,55 @@ impl P2P {
                 Ok(())
             }
             Command::ConnectToPeer(connect_to_peer_command) => {
-                // Whitelist peer
                 self.swarm
                     .behaviour_mut()
                     .allow_list
                     .allow_peer(connect_to_peer_command.peer_id);
 
-                // Add the peer to our config lists.
                 self.config.allowlist.push(connect_to_peer_command.peer_id);
-                self.config
-                    .connect_to
-                    .push(connect_to_peer_command.peer_addr.clone());
-
-                // Add peer to swarm
-                self.swarm.add_peer_address(
-                    connect_to_peer_command.peer_id,
-                    connect_to_peer_command.peer_addr.clone(),
-                );
+                for addr in &connect_to_peer_command.peer_addrs {
+                    self.config.connect_to.push(addr.clone());
+                    self.swarm
+                        .add_peer_address(connect_to_peer_command.peer_id, addr.clone());
+                }
+
+                let mut sorted_addrs = connect_to_peer_command.peer_addrs.clone();
+                sorted_addrs.sort_by_key(|a| {
+                    let s = a.to_string();
+                    if s.contains("/quic") { 0 } else { 1 }
+                });
```

Comment:
This is bad.

Please use [`Multiaddr::protocol_stack()`](https://docs.rs/libp2p/latest/libp2p/struct.Multiaddr.html#method.protocol_stack).

Also you can do this in one iteration instead of two that you are doing here with the `for` loop above.